docs/development.txt

```diff
@@ -0,0 +1,82 @@
+[[development-process]]
+JanusGraph Development Process
+==============================
+
+The following sections describe the JanusGraph development process.
+
+
+[[development-decisions]]
+Development Decisions
+---------------------
+
+Many development decisions will be made during the day-to-day work of JanusGraph contributors without any extra
+process overhead. However, for larger bodies of work and significant updates and additions, the following
+process shall be followed. Significant work may include, but is not limited to:
+
+* A major new feature or subproject eg. a new storage adapter
+* Incrementing the version of a core dependency such as a Apache TinkerPop
+* Addition or deprecation of a public API
+* Internal shared data-structure or API change
+
+For these sorts of changes, contributors will:
+
+* Create one or more issues in the Github issue tracker
+* Start a [DISCUSS] thread on the https://groups.google.com/forum/#!forum/janusgraph-dev[janusgraph-dev] list where the proposed change may be discussed by committers
+and other community members
+* When the proposer feels it appropriate, a VOTE shall be called. As with pull requests three +1 votes, including
+one from the relevant Technical Steering Committee (TSC) member if the change is limited to a single project, are required for the change to be accepted.
+
+Though not required, it is strongly recommended that contributors limit the amount of development work they
+perform before their change is accepted. In the instance where the change is not approved or significant updates to
+the proposal are made, we wish to minimize the amount of rework required.
```

Comment:
I think this is superfluous upon rereading so I'm taking it out.  My intent was to discourage folks from spending a ton of time on development before seeking consensus on whether or not a change is a good idea or not but I think the process makes that clear already.